### PR TITLE
Feature/8395 qrscanner/badge notification start tab

### PR DESCRIPTION
**Pull request by naveeddotio**
## Description
Adding the functionality to show a badge on start tab and respective unit tests.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9395

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift [hunk 1]
```diff
@@ -130,6 +131,8 @@ class CoronaTestService {
 					Log.info("[CoronaTestService] PCR test registered: \(private: String(describing: self?.pcrTest), public: "PCR Test result")", log: .api)
 
 					Analytics.collect(.testResultMetadata(.registerNewTestMetadata(Date(), registrationToken, .pcr)))
+					// updating badge count for home tab
+					self?.unseenTestsCount.value += 1
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift:135]: Can we add a parameter `markAsUnseen` to this func so we don't add a badge count if the user adds their test from the home screen? (We have the same thing for health certificates already in place)

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift [hunk 2]
```diff
@@ -183,6 +187,9 @@ class CoronaTestService {
 					Analytics.collect(.testResultMetadata(.updateTestResult(.positive, registrationToken, .pcr)))
 					Analytics.collect(.keySubmissionMetadata(.submittedWithTeletan(true, .pcr)))
 					
+					// updating badge count for home tab
+					self?.unseenTestsCount.value += 1
+
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift:192]: There is no way currently to add a teletan from the universal scanner so I think it should not be required to mark it as unseen here.

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift [hunk 3]
```diff
@@ -267,6 +274,9 @@ class CoronaTestService {
 
 					Analytics.collect(.testResultMetadata(.registerNewTestMetadata(Date(), registrationToken, .antigen)))
 
+					// updating badge count for home tab
+					self?.unseenTestsCount.value += 1
+
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift:279]: Here I also suggest to add the `markAsUnseen` parameter to the func

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift (hunk 1 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift:135]: Make sense

**Review by 30mar — APPROVED** [review]

**Review by nickguendling — APPROVED** [review]

**Pull request merged**